src/terminal/parser/OutputStateMachineEngine.cpp

```diff
@@ -151,14 +156,31 @@ bool OutputStateMachineEngine::ActionPrintString(const std::wstring_view string)
         return true;
     }
 
+    // Skip Unicode directional isolate characters (U+2066..U+2069).
+    std::wstring filtered;
+    filtered.reserve(string.size());
+    for (const auto wch : string)
+    {
+        if (wch >= wchar_t(0x2066) && wch <= wchar_t(0x2069))
+        {
+            continue;
+        }
+        filtered.push_back(wch);
+    }

```

Comment:
Performance is not really my area, but I'm slightly concerned that we have to allocate a new buffer here for every run of string output. I thought maybe we could make the passed in string mutable and then just delete the unwanted characters from the original source, but that feels very messy, and I'm not sure if that would be a safe thing to do. So for now, maybe leave it as is, and wait to see what others have to say.

Reply:
Resolving as no longer relevant with current implementation.